toolchain/sem_ir/function.cpp

```diff
@@ -35,14 +38,18 @@ auto GetCalleeFunction(const File& sem_ir, InstId callee_id) -> CalleeFunction {
   }
 
   result.function_id = fn_type->function_id;
+  result.instance_id = fn_type->instance_id;
   return result;
 }
 
-auto Function::declared_return_type(const File& file) const -> TypeId {
+auto Function::declared_return_type(const File& file,
+                                    GenericInstanceId specific_id) const
+    -> TypeId {
   if (!return_storage_id.is_valid()) {
     return TypeId::Invalid;
   }
-  return file.insts().Get(return_storage_id).type_id();
+  return GetTypeInInstance(file, specific_id,
+                           file.insts().Get(return_storage_id).type_id());
```

Comment:
To note, the accessor naming is starting to feel awkward here. This call seems non-trivial.

Reply:
Renamed away from accessor style.